torchrl/modules/models/rlhf.py

```diff
@@ -0,0 +1,94 @@
+from pathlib import Path
+
+import torch
+from torch import nn as nn
+from transformers import GPT2LMHeadModel, GPT2TokenizerFast
+
+
+class GPT2RewardModel(nn.Module):
+    """Wrapper around GPT2-like models to enable their use as reward models.
+
+    This wrapper replaces the language modelling head of the GPT2 model with a new
+    linear layer with 1 output that can be used as a reward signal. It also exposes the
+    method ``compute_reward_loss`` which calculates the reward loss by comparing two
+    batches, one chosen, one rejected.
+    """
+
+    def __init__(self, model_path=None):
+        super().__init__()
+        if model_path:
+            model = GPT2LMHeadModel.from_pretrained(model_path, return_dict=False)
+        else:
+            model = GPT2LMHeadModel(GPT2LMHeadModel.config_class())
+
+        self.config = model.config
+        self.transformer = model.transformer
+
+        # replace last layer with the reward layer
+        self.lm_head = nn.Linear(self.config.n_embd, 1, bias=False)
+        self.PAD_ID = GPT2TokenizerFast.from_pretrained("gpt2").eos_token_id
```

Comment:
shouldn't this be parametrized? the different gpt2 flavours have different tokenizer
```suggestion
        self.PAD_ID = GPT2TokenizerFast.from_pretrained(model.name_or_path).eos_token_id
```
Note: this means we should probably also store the tokenizer?

Reply:
> the different gpt2 flavours have different tokenizer

They do? I was under the impression it was a single encoding scheme for all variants and it was the number of parameters that changes?

Reply:
double checked. the comparison fails but the underlying data looks like to be the same.